cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/IncentiveSystem.java

```diff
@@ -0,0 +1,83 @@
+package org.cryptimeleon.incentivesystem.cryptoprotocol;
+
+import org.cryptimeleon.craco.sig.sps.eq.SPSEQSignature;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.IncentivePublicParameters;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.Token;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderKeyPair;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserKeyPair;
+import org.cryptimeleon.math.structures.rings.zn.Zn;
+
+/*
+ * Contains all main algorithms of the T2 incentive system.
+ */
+public class IncentiveSystem {
+
+    // Public parameters
+    IncentivePublicParameters pp;
+
+    public IncentiveSystem(IncentivePublicParameters pp) {
+        this.pp = pp;
+    }
+
+    public static IncentivePublicParameters setup() {
+        return Setup.trustedSetup(Setup.PRF_KEY_LENGTH);
```

Comment:
why is setup its own class?

Reply:
i  suppose it capsulated shared code

Reply:
I guess it is just some historical artifact. I can move it to the InvertiveSystem class if you prefer that

Reply:
actually it leave it this way. It simplifies the Incentive system class.